adam/perception/_matcher.py

```diff
@@ -893,67 +893,40 @@ def __init__(self, GM: GraphMatching, graph_node=None, pattern_node=None) -> Non
             # Now we add every other node...
 
             # Updates for T_1^{in}
-            # we use immutableset to guarantee deterministic iteration
-            new_nodes_0: ImmutableSet[Any] = immutableset(
-                flatten(
-                    [
-                        predecessor
-                        for predecessor in GM.graph.predecessors(node)
-                        if predecessor not in GM.graph_node_to_pattern_node
-                    ]
-                    for node in GM.graph_node_to_pattern_node
-                )
-            )
-            for node in new_nodes_0:
-                if node not in GM.graph_nodes_in_or_preceding_match:
-                    GM.graph_nodes_in_or_preceding_match[node] = self.depth
+            for node in GM.graph_node_to_pattern_node:
+                for predecessor in GM.graph.predecessors(node):
+                    if (
+                        predecessor not in GM.graph_node_to_pattern_node
+                        and node not in GM.graph_nodes_in_or_preceding_match
```

Comment:
Should this be:
```suggestion
                        and predecessor not in GM.graph_nodes_in_or_preceding_match
```

Reply:
Hm, possibly, but isn't that what it was doing before?
[Line 908](https://github.com/isi-vista/adam/pull/907/files#diff-055afb1c07a0ca7f2989c5ab842afcefL908).

Reply:
@spigo900 : The naming here is confusing. Note that `node` on line 908 is coming from `new_nodes_0` and therefore corresponds to `precedessor` (and not `node`) from the old list comprehension.

Reply:
Ah, I see. Thanks!

Reply:
@gabbard That fixed it.